packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -556,46 +544,91 @@ - (NSDictionary *)encodingMap
            nil);
     return;
   }
-  NSString *path = localUri.path;
+  if (![self _checkHeadersDictionary:options[@"headers"]]) {
+    reject(@"ERR_FILESYSTEM_INVALID_HEADERS",
+           @"Invalid headers dictionary. Keys and values should be strings.",
+           nil);
+    return;
+  }
+
+  NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+  NSDictionary *headers = options[@"headers"];
+  if (headers != nil) {
+    for (NSString *headerKey in headers) {
+        [request setValue:[headers valueForKey:headerKey] forHTTPHeaderField:headerKey];
+    }
+  }
   
-  NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
-  sessionConfiguration.requestCachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
-  NSDictionary *headerDict = (NSDictionary *) [options objectForKey:@"headers"];
-  if (headerDict != nil) {
-    sessionConfiguration.HTTPAdditionalHeaders = headerDict;
+  EXFileSystemSessionType type = [self _importSessionType:options[@"sessionType"]];
+  NSURLSessionDownloadTask *task = [[self _sessionForType:type] downloadTaskWithRequest:request];
+  EXSessionTaskDelegate *exTask = [[EXSessionDownloadTaskDelegate alloc] initWithResolve:resolve
```

Comment:
Would `taskDelegate` variable name work too? `exTask` is more difficult to discern (with `task` nearby)